### PR TITLE
Bring up onboarding screen on token expire

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: src/app/core/guard/workspaces.guard.ts [hunk 1]
```diff
@@ -32,19 +34,28 @@ export class WorkspacesGuard implements CanActivate {
...
         this.settingsService.getGeneralSettings(workspaceId),
+        this.settingsService.getOrganisations(workspaceId)
       ]
     ).pipe(
       map(response => !!response),
       catchError(error => {
         const that = this;
+
+        if (error.status === 400 && error.error.message === 'Xero connection expired') {
+          that.settingsService.deleteXeroCredentials(workspaceId).subscribe(() => {
```
Comment (Shwetabhk) [inline src/app/core/guard/workspaces.guard.ts:45]: Does delete API already exist?

File: src/app/core/guard/workspaces.guard.ts [hunk 2]
```diff
@@ -32,19 +34,28 @@ export class WorkspacesGuard implements CanActivate {
...
+            that.authService.logout();
```
Comment (Shwetabhk) [inline src/app/core/guard/workspaces.guard.ts:46]: Why are we logging out and not just redirecting to the dashboard?
I think this is unrelated to onboarding but still?

File: src/app/core/guard/workspaces.guard.ts (hunk 1 above)
Comment (ashwin1111) [reply src/app/core/guard/workspaces.guard.ts:45]: yes

File: src/app/core/guard/workspaces.guard.ts (hunk 2 above)
Comment (ashwin1111) [reply src/app/core/guard/workspaces.guard.ts:46]: We do the same thing in qbo.
Can set onboarded to false and redirect to dashboard, even that works. Will change this and test it out.

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**